R/summarize_weighted_production.R

```diff
@@ -77,15 +137,63 @@ add_weighted_loan_production <- function(data, use_credit_limit = FALSE) {
   total_size_by_sector <- distinct_loans_by_sector %>%
     summarize(total_size = sum(.data[[loan_size]]))
 
-  data %>%
+  out <- data
+  metric_nm <- metric_col <- "production"
+  if (buildout) {
+    out <- add_buildout(out)
+    metric_nm <- "buildout"
+    metric_col <- "build_out"
+  }
+
+  out <- out %>%
     left_join(total_size_by_sector, by = "sector") %>%
     mutate(
       loan_weight = .data[[loan_size]] / .data$total_size,
-      weighted_loan_production = .data$production * .data$loan_weight
+      "weighted_loan_{metric_nm}" := .data[[metric_col]] * .data$loan_weight
```

Comment:
This is pretty cool; but support for this kind of "glue-like" strings is very recent:
https://www.tidyverse.org/blog/2020/02/glue-strings-and-tidy-eval/

So I maybe it's best to replace this clever approach with something more vanilla.

Reply:
Oh that's very cool... I love that lol

Reply:
Done. We no longer use the clever approach that would make us depend on a too-recent version of rlang.